### PR TITLE
[#155907831] Enables filtering of activities via dropdown menu

### PR DESCRIPTION
**Pull request by jmnyarega**
## What the PR does?
 - Add functionality to the drop-down menu
## Description
 - Initially, the dropdown menu could not filter the activities when a certain status is selected.
 - The PR add the functionality to enable filtering of the activities via state

## Relevant Tickets
 [PivotalTracker Link](https://www.pivotaltracker.com/n/projects/2154966/stories/155907831)
## Screenshots
screen shot 2018-05-02 at 13 15 26
screen shot 2018-05-02 at 13 15 35
screen shot 2018-05-02 at 13 15 42

### CODE REVIEW

File: src/helpers/stringFormatter.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const capitizeString = stringValue => (
```
Comment (mkiterian) [inline src/helpers/stringFormatter.js]: capitalizeString

File: src/components/header/PageHeader.jsx [hunk 2]
```diff
@@ -15,15 +15,26 @@ class PageHeader extends Component {
...
     super(props);
     this.state = {
+      selectedStatus: 'All',
       showFilterOptionsDropdown: false,
+      statuses: ['All', 'In review', 'Pending', 'Rejected', 'Approved'],
+      activeClass: 'filterOptions__option--active',
     };
   }
 
+  componentWillReceiveProps(nextProps) {
```
Comment (mkiterian) [inline src/components/header/PageHeader.jsx]: use `getDerivedStateFromProps`

File: src/containers/MyActivities.jsx [hunk 3]
```diff
@@ -40,6 +40,27 @@ class MyActivities extends Component {
...
   };
 
+  /**
+   * React component lifecycle method getDerivedStateFromProps
+   * @param {Object} nextProps - props
+   */
+  static getDerivedStateFromProps(nextProps) {
+    return {
+      activities: nextProps.myActivities,
+      filteredMyactivities: nextProps.myActivities,
```
Comment (mkiterian) [inline src/containers/MyActivities.jsx]: filteredMyActivities

File: src/helpers/filterActivities.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import capitalizeString from './stringFormatter';
+
+const filterActivities = (status, context) => {
+  const filteredActivities = context[Object.keys(context)[0]]
```
Comment (mkiterian) [inline src/helpers/filterActivities.js]: is `context[Object.keys(context)[0]]` === context.activities ?

Comment (jmnyarega) [reply src/helpers/filterActivities.js]: They will definitely fail 😢

File: src/containers/MyActivities.jsx (hunk 3 above)
Comment (jmnyarega) [reply src/containers/MyActivities.jsx]: Please color

File: src/helpers/filterActivities.js (hunk 4 above)
Comment (jmnyarega) [reply src/helpers/filterActivities.js]: Fixed, Please review @mkiterian

**Comment by jmnyarega** [comment]
I have included the tests @mkiterian

**Pull request merged**